### DISCUSSION TITLE
I am trying to implement my network an EVM on my VPS with Blockscout

### DISCUSSION THREAD

**Discussion by scolcoin** [blockscout/blockscout#10708] [Support] [answered]
**Hello,**

My name is Juan Pablo Abadia from Colombia. Thank you very much for your attention.

### Problem Description:
I am trying to install Ubuntu 20.04 with Docker Compose on my VPS but cannot establish communication with my Geth node. The node is configured with the following command:

`pm2 start "./geth -snapshot=false --datadir mnode1 --syncmode 'full' --gcmode=archive --port 40605 --http --http.addr '0.0.0.0' --http.corsdomain='*' --http.port 3545 --http.vhosts='*' --http.api 'personal,eth,net,web3,personal,admin,miner,txpool,debug,clique' --bootnodes 'enode://134fd23d8e72cf41663dcee1063306a6e43b360137e97d1729a4b654fa478e7de4f756ae133bfcd896a39bf2bc4ed6e078679a69d5248e363b4ee399bd4e6e54@185.249.227.141:0?discport=40606, enode://7212b69187bedf26b75fdec9dce7e2f9013c78590a04e7c24a09fd4a843223b6aacc6d78d1b5c4f784592bc1baf4edf9e29ff929a8457bf04ce652d6797ddf34@185.249.227.142:40605?discport=40606, enode://9f93c84d86ed8ef6342760a0c2bc5fd47255ab4fc1dbc8e064d2d11107268e295a2fb36b34b64e79ef8059794cfe5fff1a3fffc6b4eca1520c3458bd79dfca5e@161.97.115.236:40605?discport=40606, enode://412255d67f1460478f54505bb1a5bb4f7b55db7c5a9f569da9d9325db09268ec87baa9df4738ec4e5d96824a29a9d8f082e1d59dfec2c7ec310ae401742c7a8d@178.238.237.90:40605?discport=40606, enode://cad2f1f3191c5e008d6a49061bb02009d93a7b634806686a9bb181ff4a2f5a4350663a827205e2bac3881e67bb67a034a308d17f157e04de84533308e009c73e@95.111.234.144:40605?discport=40606, enode://a535db073e8cbb0b1b1eb6cb235d9e893fe5dcefc81366839b5ffcbde07b3d14b7c7f3df42c675c8e8a768e873e527f969f34cafbf06119101a0992e9110668d@178.18.254.7:40605?discport=40606' --networkid 65450 --allow-insecure-unlock" --name "start-node1" -- start`

### my network configuration is: 
https://github.com/scolcoin/ScolNetwork 

`export HOSTNAME="localhost"
export ETHEREUM_JSONRPC_HTTP_URL=http://localhost:3545
export ETHEREUM_JSONRPC_WS_URL=ws://localhost:3546
export ETHEREUM_JSONRPC_TRACE_URL=http://localhost:3545
export ETHEREUM_JSONRPC_VARIANT=geth
export MIX_ENV=prod
export COIN=SCOL
export CHAIN_ID=65450
export CURRENCY=SCOL
export NETWORK=""
export SUBNETWORK="Scolcoin Mainnet"
export BLOCK_TRANSFORMER=clique`

### look common-blockscout.env:
https://github.com/scolcoin/blockscout/blob/master/docker-compose/envs/common-blockscout.env

### look docker-compose.yml:
https://github.com/scolcoin/blockscout/blob/master/docker-compose/docker-compose.yml

### look error:
`backend           | {"time":"2024-09-03T23:13:24.356Z","severity":"error","message":"Task #PID<0.6818.0> started from Indexer.Block.Realtime.Fetcher terminating\n** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 113ms. This means requests are coming in an^Ction: \"http://host.docker.internal:3545/\", debug_traceBlockByNumber: \"http://host.docker.internal:3545/\"], http_options: [recv_timeout: 600000, timeout: 600000, hackney: [pool: :ethereum_jsonrpc]]], variant: EthereumJSONRPC.Geth], receipts_batch_size: 250, receipts_concurrency: 10}, false]","metadata":{"error":{"initial_call":null,"reason":"** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 116ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:\n\n  1. Ensuring your database is available and that you can connect to it\n  2. Tracking down slow queries and making sure they are running fast enough\n  3. Increasing the pool_size (although this increases resource consumption)\n  4. Allowing requests to wait longer by increasing :queue_target and :queue_interval\n\nSee DBConnection.start_link/2 for more information\n\n    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1054: Ecto.Adapters.SQL.raise_sql_call_error/1\n    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:952: Ecto.Adapters.SQL.execute/6\n    (ecto 3.11.2) lib/ecto/repo/queryable.ex:232: Ecto.Repo.Queryable.execute/4\n    (explorer 6.8.0) lib/explorer/chain/block.ex:506: Explorer.Chain.Block.set_refetch_needed/1\n    (explorer 6.8.0) lib/explorer/chain/import.ex:366: Explorer.Chain.Import.set_refetch_needed_for_partially_imported_blocks/1\n    (explorer 6.8.0) lib/explorer/chain/import.ex:335: Explorer.Chain.Import.insert_runner_to_changes_list/2\n    (explorer 6.8.0) lib/explorer/chain/import.ex:131: Explorer.Chain.Import.all/1\n    (indexer 6.8.0) lib/indexer/block/realtime/fetcher.ex:229: Indexer.Block.Realtime.Fetcher.import/2\n"}}}`

**I can't connect the blockchain with blokscout dockers please helpme!**

### Refe:
**docker-compose ps:**
`NAME               IMAGE                                        COMMAND                  SERVICE            CREATED          STATUS                         PORTS

backend            blockscout/blockscout:latest                 "sh -c 'bin/blocksco…"   

backend            20 minutes ago   Up 20 minutes                  
db                 postgres:15                                  "docker-entrypoint.s…"   db                 20 minutes ago   Up 20 minutes (healthy)        0.0.0.0:7432->5432/tcp, :::7432->5432/tcp

frontend           ghcr.io/blockscout/frontend:latest           "./entrypoint.sh nod…"   

frontend           20 minutes ago   Up 20 minutes                  3000/tcp

proxy              nginx                                        "/docker-entrypoint.…"   proxy              20 minutes ago   Up 20 minutes                  0.0.0.0:80->80/tcp, :::80->80/tcp, 0.0.0.0:8080-8081->8080-8081/tcp, :::8080-8081->8080-8081/tcp

redis-db           redis:alpine                                 "docker-entrypoint.s…"   

redis-db           20 minutes ago   Up 20 minutes                  6379/tcp

sig-provider       ghcr.io/blockscout/sig-provider:latest       "./sig-provider-serv…"   sig-

provider       20 minutes ago   Up 20 minutes                  

stats              ghcr.io/blockscout/stats:latest              "./stats-server"         stats              20 minutes ago   Up 20 minutes                  

stats-db           postgres:15                                  "docker-entrypoint.s…"   

stats-db           20 minutes ago   Up 20 minutes (healthy)        0.0.0.0:7433->5432/tcp, :::7433->5432/tcp

user-ops-indexer   ghcr.io/blockscout/user-ops-indexer:latest   "./user-ops-indexer-…"   user-

ops-indexer   20 minutes ago   Restarting (1) 9 seconds ago   

visualizer         ghcr.io/blockscout/visualizer:latest         "./visualizer-server"    

visualizer         20 minutes ago   Up 20 minutes `

**Comment by GimluCom** [comment]
Please use `host.docker.internal` instead of `0.0.0.0` in your [`docker-compose.yml`](https://github.com/scolcoin/blockscout/blob/1ddca3461f5daf9e9827bc4e52b30035ce180951/docker-compose/docker-compose.yml#L44-L46)

If you use websockets, please ensure you have set these arguments on your geth node:
`--ws --ws.port 3546 --ws.addr 0.0.0.0`

**Comment by scolcoin** [comment]
update: node Scolcoin ./geth -snapshot=false --datadir mnode1 --syncmode 'full' --gcmode=archive --port 40605 --http --http.addr '0.0.0.0' --http.corsdomain='*' --http.port 3545 --http.vhosts='*' --http.api 'personal,eth,net,web3,personal,admin,miner,txpool,debug,clique' --bootnodes 'enode://134fd23d8e72cf41663dcee1063306a6e43b360137e97d1729a4b654fa478e7de4f756ae133bfcd896a39bf2bc4ed6e078679a69d5248e363b4ee399bd4e6e54@185.249.227.141:0?discport=40606' --networkid 65450 --ws.port 3546 --ws.origins=* --ws.addr 0.0.0.0 --allow-insecure-unlock

Env:
https://github.com/scolcoin/blockscout/blob/master/docker-compose/envs/common-blockscout.env
example: ETHEREUM_JSONRPC_TRACE_URL=http://host.docker.internal:3545/

error
![image](https://github.com/user-attachments/assets/328c2343-39e9-4977-be12-9b82657cc9f7)
![image](https://github.com/user-attachments/assets/60cd528b-8f00-4edf-b891-9ab642b97ef7)

**Comment by scolcoin** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/dca8f2f6-f682-4f00-a4cb-ebb09b164d51)
I think the error was because at the beginning I had my local geth, that is, ./geth, then I passed it to bin sudo mv geth /usr/local/bin/ and now it is reading the blockchain
